### PR TITLE
Validator

### LINKED ISSUE #2921: Validate `.data[].secretKey` duplicates
**Describe the bug**
I have same `secretKey` `SERVICE_NAME` but point to diferenet parameters in AWS Parameter Store. I see that operator take the last one in YAML and inject into the secret. So in my example into `test` secret I will go `SERVICE_NAME` from ` /dev/project_env_vars` Parameter instead of `/dev/shared_env_vars` Parameter. 

**To Reproduce**
I have such Externalsecret
```yaml
apiVersion: external-secrets.io/v1alpha1
kind: ExternalSecret
metadata:
  name: test
  namespace: default
spec:
  secretStoreRef:
    name: ssm-secret-store-global
    kind: ClusterSecretStore
  refreshInterval: "3m"
  target:
    name: test-env
    creationPolicy: 'Owner'
  data:
    - secretKey: SERVICE_NAME
      remoteRef:
        key: /dev/shared_env_vars
        property: TEST_URL
    - secretKey: KAFKA_BROKER_0
      remoteRef:
        key: /dev/shared_env_vars
        property: KAFKA_BROKER_0
    - secretKey: KAFKA_BROKER_1
      remoteRef:
        key: /dev/shared_env_vars
        property: KAFKA_BROKER_1
    - secretKey: KAFKA_BROKER_2
      remoteRef:
        key: /dev/shared_env_vars
        property: KAFKA_BROKER_2
    - secretKey: SERVICE_NAME
      remoteRef:
        key: /dev/project_env_vars
        property: SERVICE_NAME
```

**Expected behavior**
Error. I want to see some kind of duplication error. This is not transperent behaviour because you can inject wrong parameter.

Operator version: **0.9.3**

### PR DESCRIPTION
**Pull request by Mehrbod2002**
## Problem Statement

Fix duplicates keys .

## Related Issue

Fixes #2921 

## Checklist

- [x] I have read the [contribution guidelines](https://external-secrets.io/latest/contributing/process/#submitting-a-pull-request)
- [x] All commits are signed with `git commit --signoff`
- [x] My changes have reasonable test coverage
- [x] All tests pass with `make test`
- [x] I ensured my PR is ready for review with `make reviewable`

### CODE REVIEW

File: apis/externalsecrets/v1beta1/externalsecret_validator.go [hunk 1]
```diff
@@ -63,5 +63,15 @@ func validateExternalSecret(obj runtime.Object) (admission.Warnings, error) {
 		}
 	}
 
+	seenKeys := make(map[string]bool)
```
Comment (shuheiktgw) [inline apis/externalsecrets/v1beta1/externalsecret_validator.go:66]: nit, we usually use `map[string]struct{}` to express a set in Go

**Review by shuheiktgw — CHANGES_REQUESTED** [review]
Thanks for your contribution, @Mehrbod2002! I left one comment, but overall it looks good to me. The DCO job fails so please take a look at [the detail ](https://github.com/external-secrets/external-secrets/pull/2929/checks?check_run_id=19336986032)and resolve it by signing the commits off.

**Comment by shuheiktgw** [comment]
https://github.com/external-secrets/external-secrets/issues/2921#issuecomment-1839411106
Would you take a look at the PR as well, @gusfcarvalho? Thanks!

**Comment by Mehrbod2002** [comment]
What do you think ? @shuheiktgw

**Comment by Mehrbod2002** [comment]
@shuheiktgw I think we should better close it .

**Comment by shuheiktgw** [comment]
Waiting for https://github.com/external-secrets/external-secrets/pull/2929#issuecomment-1841823953

**Comment by gusfcarvalho** [comment]
Sorry for the delay! This logic would not work for many use cases with creationPolicy=Merge and DeletionPolicy=Merge/Delete, where users effectively leverage the duplicate keys to create `default/overlay` configuration.

this logic should only be applied to DeletionPolicy=Retain in order for it to be accurate.

**Comment by Mehrbod2002** [comment]
@shuheiktgw  @gusfcarvalho can you review it ?

**Comment by Mehrbod2002** [comment]
@shuheiktgw  @gusfcarvalho can you review it ?

**Comment by shuheiktgw** [comment]
Thanks, @gusfcarvalho, and I'm sorry I cannot review this PR for a while. I have been considering other edge cases if the SecretKey could be duplicated. For example, if users use the CreationPolicy = Merge, it could be duplicated. Also, some providers support the property field, and IIUC, the key could be duplicated as well in that case. Considering those complexities, I have to disagree with setting validation there. There is too little to gain here... If you have any objections, please feel free to reopen it, but I'm closing this PR for now 🙇

**Comment by Dgadavin** [comment]
@shuheiktgw Good day. I agree that SecretKey could be duplicated but the logic is not clear. As I mentioned in issue that I opened when you add two SecretKey with different values it takes the last one without any log message or k8s event or other things that can help you to debug. And in real life(production) this can create a real troubels.

**Comment by shuheiktgw** [comment]
Hi @Dgadavin, thanks for pointing it out! Hmm, now that I read my comment above again, I feel something is off. I may have mixed up PushSecret with ExternalSecret at that time... I cannot reopen this PR since the original branch has been deleted, but let me cherry-pick the commits and create a new PR 🙇  @Mehrbod2002

**Comment by Mehrbod2002** [comment]
@shuheiktgw Can i rewrite in new PR ?

**Comment by shuheiktgw** [comment]
That would be great. Thank you! @Mehrbod2002

**Pull request closed without merging**